modules/login/login_module.e

```diff
@@ -0,0 +1,1037 @@
+			a_response.subscribe_to_block_hook (Current)
+			a_response.subscribe_to_value_table_alter_hook (Current)
+		end
+
+feature -- Hooks
+
+	value_table_alter (a_value: CMS_VALUE_TABLE; a_response: CMS_RESPONSE)
+			-- <Precursor>
+		do
+			if attached current_user (a_response.request) as l_user then
+				a_value.force (l_user, "user")
+			end
+		end
+
+	menu_system_alter (a_menu_system: CMS_MENU_SYSTEM; a_response: CMS_RESPONSE)
+			-- Hook execution on collection of menu contained by `a_menu_system'
+			-- for related response `a_response'.
+		local
+			lnk: CMS_LOCAL_LINK
+		do
+			if attached a_response.current_user (a_response.request) as u then
+				create lnk.make (u.name +  " (Logout)", "account/roc-logout" )
+			else
+				create lnk.make ("Login", "account/roc-login")
+			end
+			a_menu_system.primary_menu.extend (lnk)
+			lnk.set_weight (98)
+		end
+
+	block_list: ITERABLE [like {CMS_BLOCK}.name]
+		local
+			l_string: STRING
+		do
+			Result := <<"login","register","reactivate","new_password", "reset_password">>
+			create l_string.make_empty
+			across Result as ic loop
+					l_string.append (ic.item)
+					l_string.append_character (' ')
+				end
+			write_debug_log (generator + ".block_list:" + l_string )
+		end
+
+	get_block_view (a_block_id: READABLE_STRING_8; a_response: CMS_RESPONSE)
+		local
+			vals: CMS_VALUE_TABLE
+		do
+			if
+				a_block_id.is_case_insensitive_equal_general ("login") and then
+				a_response.request.path_info.starts_with ("/account/roc-login")
+			then
+				get_block_view_login (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("register") and then
+				a_response.request.path_info.starts_with ("/account/roc-register")
+			then
+				get_block_view_register (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("reactivate") and then
+				a_response.request.path_info.starts_with ("/account/reactivate")
+			then
+				get_block_view_reactivate (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("new_password") and then
+				a_response.request.path_info.starts_with ("/account/new-password")
+			then
+				get_block_view_new_password (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("reset_password") and then
+				a_response.request.path_info.starts_with ("/account/reset-password")
+			then
+				get_block_view_reset_password (a_block_id, a_response)
+			end
+		end
+
+	handle_login (api: CMS_API; req: WSF_REQUEST; res: WSF_RESPONSE)
+		local
+			r: CMS_RESPONSE
+			br: BAD_REQUEST_ERROR_CMS_RESPONSE
+		do
+			create {GENERIC_VIEW_CMS_RESPONSE} r.make (req, res, api)
+			r.set_value ("Login", "optional_content_type")
+			r.execute
+		end
+
+	handle_workaround_filter (api: CMS_API; req: WSF_REQUEST; res: WSF_RESPONSE)
+		local
+			r: CMS_RESPONSE
+			br: BAD_REQUEST_ERROR_CMS_RESPONSE
+		do
+			create {GENERIC_VIEW_CMS_RESPONSE} r.make (req, res, api)
```

Comment:
unused `br' local
I would rename br as r
and have r: BAD_REQUEST_ERROR_CMS_RESPONSE 
and then create r.make ....